ksql-common/src/main/java/io/confluent/ksql/util/SchemaUtil.java

```diff
@@ -324,25 +326,40 @@ public static String getSqlTypeName(final Schema schema) {
   private static String getStructString(final Schema schema) {
     return schema.fields().stream()
         .map(field -> field.name() + " " + getSqlTypeName(field.schema()))
-        .collect(Collectors.joining(", ", "STRUCT <", ">"));
+        .collect(Collectors.joining(", ", "STRUCT<", ">"));
   }
 
   static org.apache.avro.Schema buildAvroSchema(final Schema schema, final String name) {
+    return buildAvroSchema(DEFAULT_NAMESPACE, name, schema);
+  }
+
+  private static org.apache.avro.Schema buildAvroSchema(
+      final String namespace,
+      final String name,
+      final Schema schema
+  ) {
     final FieldAssembler<org.apache.avro.Schema> fieldAssembler = org.apache.avro.SchemaBuilder
-        .record(name).namespace("ksql")
+        .record(name).namespace(namespace)
         .fields();
 
     for (final Field field : schema.fields()) {
+      final String fieldName = field.name().replace(".", "_");
```

Comment:
Embedded struct's are converted to in-line Avro records. Avro records need a name and the namespace + name need to be unique.  To accomplish this, the code build up a unique namespace for each struct.  The tests cover this.